mldsa/poly.c

```diff
@@ -320,6 +327,16 @@ uint32_t mld_poly_chknorm(const mld_poly *a, int32_t B)
     invariant((t == 0) == array_abs_bound(a->coeffs, 0, i, B))
   )
   {
+    /*
+     * Since we know that a is in [-REDUCE32_RANGE_MAX+1, REDUCE32_RANGE_MAX-1],
+     * and B <= MLDSA_Q - REDUCE32_RANGE_MAX, to check if a mod+- MLDSA_Q is
+     * in [-B+1, B-1], it suffices to check if a is in [-B+1, B-1].
+     * In other words: If a is not in [-B+1, B-1], a +- k*MLDSA_Q for k >= 1 is
+     * also not in [-B+1, B-1].
+     */
+    mld_assert(a->coeffs[i] < B || a->coeffs[i] - MLDSA_Q <= -B);
```

Comment:
It might be worth adding a comment here that you're trying to write
`
a ==> b
`
but need to write
`
!a || b
`
to make it executable, right?